### PR TITLE
vdk-heartbeat: cover requirements.txt automatic installs

### PR DESCRIPTION
**Pull request by tozka**
Upon `vdk deploy` the Control Service is responsible for installing all external dependecies defined in requirements.txt of the data job so they can be used during a cloud run.

A recent outage in one of our customers caused data jobs to ignore their requirements.txt and not install any external libraries. It was trace to be due to a release of [data job base
image](https://hub.docker.com/layers/versatiledatakit/data-job-base-python-3.11)

CICD of the open source VDK did not catch this. 

I am extending VDK heartbeat to test for the issue so we can catch it in the future.
 
See also https://github.com/vmware/versatile-data-kit/issues/2391

### CODE REVIEW

File: projects/vdk-heartbeat/.gitlab-ci.yml [hunk 1]
```diff
@@ -28,6 +28,28 @@ vdk-heartbeat-test:
...
 
 
+vdk-heartbeat-release-acceptance-test:
+  stage: pre_release
+  before_script:
+    - cd projects/vdk-heartbeat
+  script:
+    - set -x
+    - python setup.py sdist --formats=gztar
+    - pip install vdk-heartbeat --extra-index-url $PIP_EXTRA_INDEX_URL
```
Comment (DeltaMichael) [inline projects/vdk-heartbeat/.gitlab-ci.yml:38]: Why are we installing heartbeat from a private repo?

File: projects/vdk-heartbeat/.gitlab-ci.yml [hunk 2]
```diff
@@ -28,6 +28,28 @@ vdk-heartbeat-test:
...
+    - export VDK_HEARTBEAT_API_TOKEN=$VDK_API_TOKEN
+    - export VDK_HEARTBEAT_OP_ID="vdk-heartbeat-$CI_PIPELINE_ID"
+    - export JOB_NAME=vdk-heartbeat-release-test-job-$(date +%s)
+    - vdk-heartbeat -f cicd/cicd_heartbeat_config.ini
+  artifacts:
+    when: always
+    reports:
+      junit: tests.xml
+  only:
+    refs: # TODO switch to "main" so it is used as a pre-release test not pre-merge test
```
Comment (DeltaMichael) [inline projects/vdk-heartbeat/.gitlab-ci.yml:48]: It's hard to understand this TODO. Should this change trigger on merge to main, on PRs or on both?

**Review by DeltaMichael — APPROVED** [review]

File: projects/vdk-heartbeat/.gitlab-ci.yml (hunk 1 above)
Comment (tozka) [reply projects/vdk-heartbeat/.gitlab-ci.yml:38]: what private repo ?

File: projects/vdk-heartbeat/.gitlab-ci.yml (hunk 2 above)
Comment (murphp15) [reply projects/vdk-heartbeat/.gitlab-ci.yml:48]: he has that change to test and is going to make changes before merging.

**Comment by murphp15** [comment]
Is this PR just a test or a fix for the issue too? 
I dont see a fix for the issue?

**Comment by tozka** [comment]
> Is this PR just a test or a fix for the issue too? I dont see a fix for the issue?

It is only a test. But it fails to reproduce the issue. So I am trying to figure out why ...

 It's still valuable coverage. So likely I will keep it.

**Comment by tozka** [comment]
> Is this PR just a test or a fix for the issue too? I dont see a fix for the issue?

The fix is in https://github.com/vmware/versatile-data-kit/pull/2429

File: projects/vdk-heartbeat/src/vdk/internal/heartbeat/vdk_heartbeat_data_job/trino/requirements.txt [hunk 3]
```diff
@@ -1 +1,2 @@
+Faker
```
Comment (doks5) [inline projects/vdk-heartbeat/src/vdk/internal/heartbeat/vdk_heartbeat_data_job/trino/requirements.txt:1]: Isn't vdk-trino considered "external" dependency, anyways?

**Review by doks5 — APPROVED** [review]

File: projects/vdk-heartbeat/src/vdk/internal/heartbeat/vdk_heartbeat_data_job/trino/requirements.txt (hunk 3 above)
Comment (tozka) [reply projects/vdk-heartbeat/src/vdk/internal/heartbeat/vdk_heartbeat_data_job/trino/requirements.txt:1]: Yes. But I am adding Faker to explicitly check for that.

**Comment by DeltaMichael** [comment]
> It is only a test. The fix(es) would be separate PR(s). But it fails to reproduce the issue. So I am trying to figure out why ...

Wait, I'm confused now. 😄 If you're just testing something, please mark it as draft.

**Comment by tozka** [comment]
> > It is only a test. The fix(es) would be separate PR(s). But it fails to reproduce the issue. So I am trying to figure out why ...
> 
> Wait, I'm confused now. 😄 If you're just testing something, please mark it as draft.

No, I mean , this is not fixing the issue , The fix is in separate PR (#2429).
 I am extending VDK heartbeat to test for the issue so we can catch it in the future.

**Pull request merged**